src/payments/views.py

```diff
@@ -0,0 +1,199 @@
+        try:
+            payment_intent = PaymentService.confirm_payment_intent(payment_intent_id=payment.payment_id)
+            payment.status = 'PAID'
+            payment.save()
+            context = {
+                'id': payment_intent.id,
+                'amount': payment_intent.amount,
+                'amount_received': payment_intent.amount_received,
+                'created': payment_intent.created,
+                'description': payment_intent.description,
+                'status': payment_intent.status,
+                'currency': payment_intent.currency,
+                'customer': payment_intent.customer
+            }
+            return Response(data=context, status=status.HTTP_200_OK)
+        except Exception as e:
+            return Response({'error_msg': str(e)}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class CancelPaymentIntent(APIView):
+
+    def post(self, request):
+        payment_id = request.query_params.get("payment_id")
+        payment = Payments.objects.get(id=payment_id)
+        if payment is None:
+            return Response({'error_msg': 'payment not found'}, status=status.HTTP_404_NOT_FOUND)
+        try:
+            payment_intent = PaymentService.cancel_payment_intent(payment_intent_id=payment.payment_id)
+            payment.delete()
+            return Response(data= {'msg': 'Delete successfully'}, status=status.HTTP_200_OK)
+        except Exception as e:
+            return Response({'error_msg': str(e)}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class RetrievePaymentIntent(APIView):    
+    def get(self, request):
+        payment_id = request.query_params.get("payment_id")
+        payment = Payments.objects.get(id=payment_id)
+        if payment is None:
+            return Response({'error_msg': 'payment not found'}, status=status.HTTP_404_NOT_FOUND)
+        try:
+            payment_intent = PaymentService.retrieve_payment_intent(payment_intent_id=payment.payment_id)
+            context = {
+                'id': payment_intent.id,
+                'amount': payment_intent.amount,
+                'created': payment_intent.created,
+                'description': payment_intent.description,
+                'status': payment_intent.status,
+                'currency': payment_intent.currency,
+                'customer': payment_intent.customer
+            }
+            return Response(data=context, status=status.HTTP_200_OK)
+        except Exception as e:
+            return Response({'error_msg': str(e)}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class PaymentsList(APIView):
+    serializer_class = PaymentSerializer
+    filter_backends = [filters.SearchFilter]
+    search_fields = ['title', 'address']
+
+    def get(self, request, format=None):
+        payments = Payments.objects.all().order_by('created_at')
+        pagesize = int(settings.PAGESIZE)
+        page_total = round(Payments.objects.all().count() / pagesize + 0.5)
```

Comment:
I think paginator will support it, please research it

Reply:
kkk ok , i will make it better